docs/storage/s3/README.md

```diff
@@ -0,0 +1,164 @@
+# Storage
+
+Description of using static object storage as database for Application Registry.
+
+## Storing Application labels
+
+There were an idea to use Apache Parquet for storing the pieces of information regarding labels.
+We want to have an easy access to the pieces of information:
+    - list all labels for application 
+    - query application by label
+    - list all labels
+
+### S3-based Solution
+
+To achieve querying by groups, the following bucket structure is suggested:
+
+![Bucket structure](./assets/bucket-structure.jpg)
+
+Data redundancy is needed to reduce requests while doing read operations.
+
+In `data.json` files we could store API, event specs and docs. However, we would need to assure that always the latest file is modified (file versioning). We can also store the files separately, but then the number of requests increases. That's why the cases below assumes that we store the data in single `data.json` file.
```

Comment:
should `data.json` file be encrypted? IMO yes, and this cause that every modification in this file requires decoding it, applying changes, encoding again and storing on S3. A lot of operation when we compare to simple `insert into` for DB.

Reply:
As far as I remember, from our set of initial meetings, we don't need to encrypt the Runtime and Application metadata, as there are no secrets, which would be held in vault or as separate, encrypted files.

However, I totally agree with you that it would be time-consuming.